lib/order.rb

```diff
@@ -0,0 +1,39 @@
+class Order
+
+  attr_reader :basket
+
+  def initialize
+    @basket = []
+  end
+
+  def add_to_basket(dish, price, quantity)
+    @selection = {}
+    @selection[:dish] = dish
+    @selection[:price] = price
+    quantity.times { @basket << @selection }
+    selection_confirmation(dish, quantity)
+  end
+
+  def selection_confirmation(dish, quantity)
+    selection_price = '%.2f' % (quantity * @selection[:price])
+    "#{quantity} x #{dish} added to your basket.\n
+That will be £#{selection_price}.\n
+Anything else?"
+  end
+
+  def calculate_total
+    @total = 0
+    @basket.each do |selection|
```

Comment:
consider having @total in the initialize to make the method shorter, and possibly using curly braces {} instead of do and end.